cmd/runtime_options.go

```diff
@@ -86,11 +88,13 @@ func getRuntimeOptions(flags *pflag.FlagSet, environment map[string]string) (lib
 		Env:                  make(map[string]string),
 	}
 
-	if envVar, ok := environment["K6_COMPATIBILITY_MODE"]; ok {
+	if envVar, ok := environment["K6_TYPE"]; ok && !opts.TestType.Valid {
 		// Only override if not explicitly set via the CLI flag
-		if !opts.CompatibilityMode.Valid {
-			opts.CompatibilityMode = null.StringFrom(envVar)
-		}
+		opts.TestType = null.StringFrom(envVar)
+	}
+	if envVar, ok := environment["K6_COMPATIBILITY_MODE"]; ok && !opts.CompatibilityMode.Valid {
```

Comment:
Out of curiosity, what is k6's "compatibility mode"?

Reply:
https://k6.io/docs/using-k6/javascript-compatibility-mode does this answer your question?

Reply:
It does! Thanks @MStoykov !